### ISSUE TITLE
Pass validators through to 'wrapped' FormControl

### ISSUE THREAD

**Issue by DataServices-TKI** [simontonsoftware/s-ng-utils#4]
(Firstly, thank you for this library, it's been super helpful so far!)

I am using the WrappedFormControlSuperclass to create a new component. I have a bunch of styling and behaviour (floating labels etc) for text input boxes that I want to encapsulate in a reusable component as our app will have hundreds of these controls and I want to keep things DRY.

It would be awesome if I could encapsulate all my validation messages etc within the custom component, so I don't have to repeat template code for common validations all over the code base.

I would try to implement this myself and submit a PR but I'm new to Angular and trying to figure out how to implement `ControlValueAccessor` and pass the validators through gave me a headache!

Please see the code below for an example of how this would be used:

## input-text.component.ts
```js
import { Component, Injector, Input } from '@angular/core';
import { WrappedFormControlSuperclass, provideValueAccessor } from 's-ng-utils';

@Component({
  selector: 'app-input-text',
  templateUrl: './input-text.component.html',
  styleUrls: ['./input-text.component.scss'],
  providers: [provideValueAccessor(InputTextComponent)],
})
export class InputTextComponent extends WrappedFormControlSuperclass<string>
{
  @Input()
  public label: string;

  constructor(injector: Injector) {
    super(injector);
  }
}
```

## input-text.component.html
```html
<div class="input-text">
  <label>
    <input class="form-control" type="text" [formControl]="formControl" />
    <span>{{ label }}</span>
  </label>
  <div *ngIf="this.formControl.errors?.required">
    {{ label }} is required!
  </div>
</div>
```

## details.component.js
```js
import { Component, OnInit } from '@angular/core';
import { FormBuilder, Validators, FormGroup } from '@angular/forms';

@Component({
  selector: 'app-details',
  templateUrl: './details.component.html',
  styleUrls: ['./details.component.scss']
})
export class DetailsComponent implements OnInit {

  detailsForm: FormGroup;

  constructor(private fb: FormBuilder) { }

  ngOnInit() {
    this.detailsForm = this.fb.group({
      email: ['', Validators.required],
      firstName: ['', Validators.required],
      lastName: ['', Validators.required],
    });
  }
}
```

## details.component.html
```html
<form [formGroup]="detailsForm">
  <app-input-text formControlName="email" label="Email"></app-input-text>
  <app-input-text formControlName="firstName" label="First Name"></app-input-text>
  <app-input-text formControlName="lastName" label="Last Name"></app-input-text>
</form>
```

EDIT: I was accidentally logged in to my company account, so mentioning my personal account so I get the update notificatons - @acodrington

**Comment by ersimont** [comment]
I'm glad you're finding the library helpful! Thank you for such a well defined issue. This is a great use case.

First, let me suggest that what you actually want is not to pass the validators set on your wrapper down to the inner form control. That only works as long as the validation can apply to both the parent and child values, but in theory those could be of completely different types. Instead I think you simply want to access the errors on the parent control.

I did some digging, and [this technique from SO](https://stackoverflow.com/a/49578414) looks like a good way to go. Here is a working example:

```ts
@Component({
  selector: "app-error-displaying-input",
  template: `
    <input [formControl]="formControl" />
    <div *ngIf="control.errors?.required">I'm required!</div>
  `,
})
export class ErrorDisplayingInputComponent extends WrappedFormControlSuperclass<
  string
> {
  constructor(@Self() public control: NgControl, injector: Injector) {
    super(injector);
    control.valueAccessor = this;
  }
}
```

I questioned whether the line `control.valueAccessor = this` is really a fully supported feature of Angular, or a hack, so I checked the video the SO user posted. Sure enough - direct from an angular core member: https://youtu.be/CD_t3m2WMM8?t=1409
